autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/ui/credential/management/viewing/SaveStateWatcher.kt

```diff
@@ -0,0 +1,140 @@
+ */
+
+package com.duckduckgo.autofill.ui.credential.management.viewing
+
+import android.text.Editable
+import android.text.TextWatcher
+import androidx.core.view.isEmpty
+import com.duckduckgo.autofill.impl.databinding.FragmentAutofillManagementEditModeBinding
+import javax.inject.Inject
+
+fun FragmentAutofillManagementEditModeBinding.watchSaveState(
+    saveStateWatcher: SaveStateWatcher,
+    saveStateUpdater: (Boolean) -> Unit = {}
+) {
+    saveStateWatcher.start(saveStateUpdater) {
+        domainTitleEditText.text.isEmpty() && domainEditText.text.isEmpty() &&
+            notesEditText.text.isEmpty() && passwordEditText.text.isEmpty() && usernameEditText.text.isEmpty()
+    }
+    domainEditText.addTextChangedListener(saveStateWatcher)
+    notesEditText.addTextChangedListener(saveStateWatcher)
+    usernameEditText.addTextChangedListener(saveStateWatcher)
+    passwordEditText.addTextChangedListener(saveStateWatcher)
+}
+
+fun FragmentAutofillManagementEditModeBinding.removeSaveStateWatcher(saveStateWatcher: SaveStateWatcher) {
+    domainTitleEditText.removeTextChangedListener(saveStateWatcher)
+    domainEditText.removeTextChangedListener(saveStateWatcher)
+    notesEditText.removeTextChangedListener(saveStateWatcher)
+    usernameEditText.removeTextChangedListener(saveStateWatcher)
+    passwordEditText.removeTextChangedListener(saveStateWatcher)
+}
+
+class SaveStateWatcher @Inject constructor() : TextWatcher {
+    private var _saveStateUpdater: (Boolean) -> Unit = {}
+    private var _allTextInputIsEmpty: () -> Boolean = { false }
+    fun start(
+        saveStateUpdater: (Boolean) -> Unit,
+        allTextInputIsEmpty: () -> Boolean
+    ) {
+        _saveStateUpdater = saveStateUpdater
+        _allTextInputIsEmpty = allTextInputIsEmpty
+    }
+
+    override fun beforeTextChanged(
+        s: CharSequence?,
+        start: Int,
+        count: Int,
+        after: Int
+    ) {
+    }
+
+    override fun onTextChanged(
+        s: CharSequence?,
+        start: Int,
+        before: Int,
+        count: Int
+    ) {
+    }
+
+    override fun afterTextChanged(s: Editable?) {
+        _saveStateUpdater(!_allTextInputIsEmpty())
+    }
+}
+
+class SaveStateWatcher1 @Inject constructor() {
+    private var _binding: FragmentAutofillManagementEditModeBinding? = null
+    private var _saveStateUpdater: (Boolean) -> Unit = {}
+    private val textWatcher = object : TextWatcher {
+        override fun beforeTextChanged(
+            s: CharSequence?,
+            start: Int,
+            count: Int,
+            after: Int
+        ) {
+        }
+
+        override fun onTextChanged(
+            s: CharSequence?,
+            start: Int,
+            before: Int,
+            count: Int
+        ) {
+        }
+
+        override fun afterTextChanged(s: Editable?) {
+            _binding?.let {
+                _saveStateUpdater(
+                    !(
+                        it.domainTitleEditText.text.isEmpty() &&
```

Comment:
Could we provide a list of EditTexts to watch instead of manually referring to each individually? Thinking if we could define the group once then there is less chance of error-prone changes in future

Reply:
This is actually unused code so I am removing this.